catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/impl/operations/ResourceOperationsOptionsTest.java

```diff
@@ -0,0 +1,379 @@
+
+    MetacardImpl metacard;
+
+    QueryRequest queryRequestMock;
+
+    @Before
+    public void setUp() throws Exception {
+        setUpFrameworkProperties();
+        setUpResourceOperationsAndMocks();
+    }
+
+    // getEnterpriseResourceOptions
+    @Test
+    public void testGetEnterpriseResourceOptions() throws Exception {
+
+        Result resultMock = mock(Result.class);
+
+        ArrayList<Result> mockResultList = new ArrayList<>();
+        mockResultList.add(resultMock);
+
+        when(queryOperationsMock.query(any(), any(), anyBoolean(), anyBoolean())).thenReturn(
+                queryResponseMock);
+        when(queryResponseMock.getResults()).thenReturn(mockResultList);
+        when(resultMock.getMetacard()).thenReturn(metacard);
+
+        resourceOperations.setQueryOperations(queryOperationsMock);
+        resourceOperations.getEnterpriseResourceOptions(metacard.getId(), fanoutEnabled);
+
+        verify(queryOperationsMock).query(any(), any(), anyBoolean(), anyBoolean());
+    }
+
+    @Test(expected = ResourceNotFoundException.class)
+    public void testGetEnterpriseResourceOptionsEmptyResults() throws Exception {
+        ArrayList<Result> mockResultList = new ArrayList<>();
+
+        when(queryOperationsMock.query(any(), any(), anyBoolean(), anyBoolean())).thenReturn(
+                queryResponseMock);
+        when(queryResponseMock.getResults()).thenReturn(mockResultList);
+
+        resourceOperations.setQueryOperations(queryOperationsMock);
+        resourceOperations.getEnterpriseResourceOptions(metacard.getId(), fanoutEnabled);
+
+        verify(queryOperationsMock).query(any(), any(), anyBoolean(), anyBoolean());
+    }
+
+    @Test(expected = ResourceNotFoundException.class)
+    public void testGetEnterpriseResourceOptionsCatchesUnsupportedQueryException()
+            throws Exception {
+
+        when(queryOperationsMock.query(any(), any(), anyBoolean(), anyBoolean())).thenThrow(
+                UnsupportedQueryException.class);
+
+        resourceOperations.setQueryOperations(queryOperationsMock);
+        resourceOperations.getEnterpriseResourceOptions(metacard.getId(), fanoutEnabled);
+
+        verify(queryOperationsMock).query(any(), any(), anyBoolean(), anyBoolean());
+    }
+
+    @Test(expected = ResourceNotFoundException.class)
+    public void testGetEnterpriseResourceOptionsCatchesFederationException() throws Exception {
+
+        when(queryOperationsMock.query(any(), any(), anyBoolean(), anyBoolean())).thenThrow(
+                FederationException.class);
+
+        resourceOperations.setQueryOperations(queryOperationsMock);
+        resourceOperations.getEnterpriseResourceOptions(metacard.getId(), fanoutEnabled);
+
+        verify(queryOperationsMock).query(any(), any(), anyBoolean(), anyBoolean());
+    }
+
+    @Test(expected = ResourceNotFoundException.class)
+    public void testGetEnterpriseResourceOptionsCatchesIllegalArgumentException() throws Exception {
```

Comment:
Is the code in this test the same code in the method `testGetEnterpriseResourceOptionsCatchesFederationException`?